### PR TITLE
Changed documentation for column names

### PR DESCRIPTION
**Pull request by turbo-ele**

### CODE REVIEW

File: blackbox/docs/sql/ddl/basics.txt [hunk 1]
```diff
@@ -120,6 +120,37 @@ length. They:
...
+    | _foo_        | _foo         |
+    +--------------+--------------+
+    | _Foo         | _foo_bar     |
+    +--------------+--------------+
+    | _fo1o        |              |
+    +--------------+--------------+
+    | __foo        |              |
+    +--------------+--------------+
+
+  - may not start with any character followed by opening and closing square
```
Comment (chaudum) [inline blackbox/docs/sql/ddl/basics.txt]: A character sequence ...

* ... followed by an opening square bracket ...
* ... followed by zero or more characters which are not closing square brackets ...
* ... followed by a closing square bracket

is not allowed as column name.

File: blackbox/docs/sql/ddl/basics.txt [hunk 2]
```diff
@@ -118,8 +118,52 @@ length. They:
...
+    +--------------+--------------+
+
+  - A character sequence ...
+
+    - ... followed by an opening squeare bracket ``[`` ...
+
+    - ... followed by zero or more characters which are not closing square
+      brackets ...
+
+    - ... followed by a closing square braket ``]`` ...
```
Comment (Autophagy) [inline blackbox/docs/sql/ddl/basics.txt]: I think this is a little confusing. I would rephrase it with something like:

```
Character sequences that conform to the subscript notation (for example `abc[abc]`) are
not allowed, as this notation is already used for the identification of columns/paths in
object and array columns.

Additionally, `information_schema.columns` returns columns that are *not* double
quoted. This would lead to conflicts when a table contains object or array columns.
```

Or something like that. Perhaps even a link to the subscript notation at https://crate.io/docs/crate/reference/sql/queries.html#object-arrays would be best as well.

File: blackbox/docs/sql/ddl/basics.txt [hunk 3]
```diff
@@ -118,8 +118,45 @@ length. They:
   - should not exceed 255 bytes when encoded with ``utf-8`` (this
     limit applies on the optionally schema-qualified table name)
 
-Column names are restricted in terms of characters. They:
+Column names are restricted in terms of characters.
 
-  - may not contain one of the following characters: ``[`` ``'`` ``]`` ``.``
+  - Columns that *start* with an underscore (``_``) followed by a lowercase
+    letter, followed by a sequence of lower case letters or underscores and
+    ending with a lower case letter are not allowed (Equivalent to
```
Comment (chaudum) [inline blackbox/docs/sql/ddl/basics.txt]: equivalent

File: blackbox/docs/sql/ddl/basics.txt [hunk 4]
```diff
@@ -118,8 +118,45 @@ length. They:
...
+    +--------------+--------------+
+    | _fo1o        |              |
+    +--------------+--------------+
+    | __foo        |              |
+    +--------------+--------------+
+
+  - Character sequences that conform to the
+    :ref:`subscript notation <sql_dql_object_arrays>` (e.g. ``col[id]``) are
+    not allowed as this notation is already used for the identification of
+    columns/paths in object and array columns.
```
Comment (chaudum) [inline blackbox/docs/sql/ddl/basics.txt]: only object columns have paths

File: blackbox/docs/sql/ddl/basics.txt [hunk 5]
```diff
@@ -118,8 +118,45 @@ length. They:
...
+
+    Additionaly, that ``information_schema.columns`` returns column names that
```
Comment (chaudum) [inline blackbox/docs/sql/ddl/basics.txt]: remove first that

Comment (chaudum) [inline blackbox/docs/sql/ddl/basics.txt]: I would put that warning to the information_schema.columns section of the documentation

File: blackbox/docs/sql/ddl/basics.txt [hunk 6]
```diff
@@ -118,8 +118,45 @@ length. They:
...
+    +--------------+--------------+
+    | _Foo         | _foo_bar     |
+    +--------------+--------------+
+    | _fo1o        |              |
+    +--------------+--------------+
+    | __foo        |              |
+    +--------------+--------------+
+
+  - Character sequences that conform to the
+    :ref:`subscript notation <sql_dql_object_arrays>` (e.g. ``col[id]``) are
```
Comment (chaudum) [inline blackbox/docs/sql/ddl/basics.txt]: col['id']

File: blackbox/docs/sql/ddl/basics.txt (hunk 4 above)
Comment (chaudum) [inline blackbox/docs/sql/ddl/basics.txt]: fields in object columns and indices of array columns

File: blackbox/docs/sql/information_schema.txt [hunk 7]
```diff
@@ -241,6 +241,10 @@ Columns Specification
 | ``table_name``                | Table Name                                    | ``String``    |
 +-------------------------------+-----------------------------------------------+---------------+
 | ``column_name``               | Column Name                                   | ``String``    |
+|                               | For fields in object columns this is not an   |               |
+|                               | identifier but a path and therefore must not  |               |
+|                               | be double quoted when programatically         |               |
```
Comment (chaudum) [inline blackbox/docs/sql/information_schema.txt]: programmatically

File: blackbox/docs/sql/information_schema.txt [hunk 8]
```diff
@@ -241,6 +241,10 @@ Columns Specification
...
+|                               | obtained                                      |               |
```
Comment (chaudum) [inline blackbox/docs/sql/information_schema.txt]: sentence should end with dot
